### DISCUSSION TITLE
Black color in graphic cosmos

### DISCUSSION THREAD

**Discussion by keyvanplays** [CosmosOS/Cosmos#2963] [General]
please help me I have black color in my kernel

**Comment by Shlepsss** [comment]
Enter Canvas.Display() at the end of graphics code part

**Reply by keyvanplays** [reply]
Is it true?
kernel.cs:
`using Cosmos.System.Graphics;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using Sys = Cosmos.System;

namespace keyvanos
{
    public class Kernel : Sys.Kernel
    {
        private Canvas canvas;
        protected override void BeforeRun()
        {
            Console.Clear();
            Console.WriteLine("welcome to keyvanos2");
            

        }

        protected override void Run()
        {
            Console.Write("Keyvanos2/$:");
            var input = Console.ReadLine();
            Console.WriteLine("your command:" + input);
            switch (input)
            {
                case "help":
                    {
                        Console.WriteLine("barai khamosh shodan type konid khamosh" );
                        Console.WriteLine("barai restart lotfan type konid restart");
                        Console.WriteLine("barai run shodan mohitgraphic gui type konid");
                        break;
                    }
                case "khamosh":
                    {
                        Console.WriteLine("dar hal khamosh shodan");
                        Sys.Power.Shutdown();
                        break;
                    }
                case "restart":
                    {
                        Console.WriteLine("dar hal restart");
                        Sys.Power.Reboot();
                        break;  
                    }
                case "clear":
                    {
                        Console.Clear();
                        break;
                    }
                case "gui":
                    {
                        Console.Clear();
                        
                        
                        gui.update();

                        break;
                    }
                default:
                    {
                        Console.WriteLine("command eshtebah ast lotfan type konid help");
                        break;
                    }
            }
        }
    }
}
`
and 
gui.cs:
`using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;
using Cosmos.System.Graphics;

namespace keyvanos
{
    class gui
    {
        private static int screenX = 800;
        private static int screenY = 640;

        private static Color[] pixelBuffer = new Color[(screenX * screenY) + screenX];
        private static Color[] pixelBufferOld = new Color[(screenX * screenY) + screenX];


        private static Canvas canvas;

        public static void init()
        {
            canvas = FullScreenCanvas.GetFullScreenCanvas();
            canvas.Mode = new Mode(screenX, screenY, ColorDepth.ColorDepth32);
        }

        public static void setPixel(int x, int y, Color c)
        {
            if (x > screenX || y > screenY) return;
            pixelBuffer[(x * y) + x] = c;
        }

        public static void drawScreen()
        {
            Pen pen = new Pen(Color.Orange);
            for (int y = 0, h = screenY; y < h; y++)
            {
                for (int x = 0, w = screenX; x < w; x++)
                {
                    if (!(pixelBuffer[(y * x) + x] == pixelBufferOld[(y * y) + x]))
                    {
                        pen.Color = pixelBuffer[(y * screenX) + x];
                        canvas.DrawPoint(pen, x, y);
                    }
                }
            }
            for (int i = 0, len = pixelBuffer.Length; i < len; i++)
            {
                pixelBuffer[i] = pixelBufferOld[i];
            }
        }


        public static void clearScreen(Color c)
        {
            for (int i = 0, len = pixelBuffer.Length; i < len; i++)
            {
                pixelBuffer[i] = c;
            }
        }

        public static void update()
        {
            clearScreen(Color.Blue);
            setPixel(1, 1, Color.Black);
            setPixel(1, 2, Color.Black);
            setPixel(2, 1, Color.Black);
            setPixel(2, 2, Color.Black);
            drawScreen();
            canvas.Display();
        }


    }
}
`
but not work
